lib/src/tag_detection.dart

```diff
@@ -818,11 +864,22 @@ class Tagger {
                     SuggestionCode.notCompatible,
                     'Package not compatible with runtime ${flutterPlatform.runtime.name} of ${flutterPlatform.name}',
                     'Because of ${LibraryGraph.formatPath(path)}')));
+
         var supports = true;
+        var hasFoundNonPrunedViolation = false;
+
         for (final lib in _topLibraries) {
-          final violationResult = violationFinder._findPlatformViolation(lib);
-          if (violationResult != null) {
-            suggestions.add(violationResult);
+          if (!hasFoundNonPrunedViolation) {
+            final violationResult = violationFinder._findPlatformViolation(lib);
+            if (violationResult != null) {
+              suggestions.add(violationResult);
+              hasFoundNonPrunedViolation = true;
+            }
+          }
```

Comment:
nit. Maybe it would be easier to read if we did this in two separate loops... or used map/where/any, like:

```dart
// Report only the first non-pruned violation as suggestion 
final firstNonPrunedViolation = _topLibraries
  .map(violationFinder._findPlatformViolation)
  .firstWhere((e) => e != null, orElse: () => null);
if (firstNonPrunedViolation != null) {
  suggestions.add(firstNonPrunedViolation);
}

// Tag is supported, if there is no pruned violations
final supports = _topLibraries
  .map(prunedViolationFinder._findPlatformViolation)
  .any((e) => e != null);
```